### PR TITLE
feat: Implement relay basics

### PR DESCRIPTION
**Pull request by jan-auer**

### CODE REVIEW

**Comment by mitsuhiko** [comment]
Before we merge this I think we should add a public key whitelist into this. While there is no danger atm in just whitelisting all relays as no confidential data is pushed I rather not end up pushing changes later that invalidate that assumption.

File: src/sentry/api/endpoints/relay_index.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+from sentry.api.base import Endpoint
+from sentry.api.paginator import OffsetPaginator
+from sentry.api.serializers import serialize
+from sentry.models import Relay
+
+
+class RelayIndexEndpoint(Endpoint):
+    permission_classes = (IsAuthenticated, )
+
+    def get(self, request):
```
Comment (mattrobenolt) [inline src/sentry/api/endpoints/relay_index.py]: Without reading much, this seems bad that we'd just return all relays to anyone logged in. I feel like we'd want this behind superuser or something.

Comment (mitsuhiko) [reply src/sentry/api/endpoints/relay_index.py]: Yeah. We should change that probably before deploy.

**Comment by jan-auer** [comment]
@HazAT `semaphore` installed, tests are failing though

**Review by mitsuhiko — APPROVED** [review]

**Pull request merged**